design-proposals/edge-infra-day2-improvements.md

```diff
@@ -0,0 +1,368 @@
+    - `STATUS_INDICATION_ERROR`: an error happened during the update.
+    - `STATUS_INDICATION_IN_PROGRESS`: unused, the progress about the 
+      update is kept into the `update_status` field of the Instance.
+    - `STATUS_INDICATION_IDLE`: the update was successful.
+  - `status_timestamp`: timestamp of the last status update. Overlaps with 
+    the `end_time`, but it's kept for consistency with Modern Status. Also, 
+    could be used in the future, if we deprecate the Instance `update_status`.
+  - `status_details`: detailed description of what happend during the 
+    update. For example, the list of packages that were installed or 
+    updated, or the reason why the Immutable OS update failed.
+  - `start_time`: timestamp of when the update job started.
+  - `end_time`: timestamp of when the update job ended.
+  - `applied_policy`: the policy that was applied during the update. 
+    Provides information of what was the target Update Policy for this update.
+
+Resource and Field Handling:
+
+- **OS Profile**:
+  - `installed_packages`: this field is handled and filled by OSRM also for 
+    Mutable OSes.
+- **Instance**:
+  - `current_os`: after provisioning, this field is handled internally by 
+    EIM. The Maintenance Manager (MM) keeps it updated after the 
+    maintenance schedule happened.
+  - `installed_packages`: this field is handled by Host Resource Manager (HRM)
+    that received the information about installed package by the Hardware 
+    Discovery Manager (HDA). For Immutable OS, this field could be filled 
+    with info from the OSProfile directly, without EN providing these 
+    information at runtime.
+  - `new_versions`: this field is handled by the OSRM, that will update 
+    it when new updates are available. 
+- **OS Update Policy**: this resource is completely handled by the User of 
+  the orchestrator. The creation is done through Northbound REST APIs. EIM 
+  won't change the content of this resource. Default OS Update Policies 
+  could be agreed and created upon Tenant creation (for example, update to 
+  latest policy). OS Update Policies cannot be updated after creation, and 
+  cannot be deleted if any OS Update Run or any Instances refers to it.
+- **OS Update Run**: this resource is created by the MM when the update 
+  job is started on the Edge Node. The content of this resource is filled by 
+  the MM, and it will be updated with the status of the update job. It is 
+  read-only from Northbound REST APIs.
+
+#### Day 2 Workflows:
+
+Day2 update workflow for Mutable and Immutable OSes:
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+    participant User
+    box Edge Infrastructure Manager
+    participant Inventory
+    participant MM as Maintenance Manager
+    end
+    box Edge Node
+    participant PUA as Platform Update Agent
+    end
+
+    rect rgb(255,255,255)
+        note over User,Inventory: User creates an OS Update Policy and schedule an update
```

Comment:
this shoudl be contextual, meaning immediately with the creation of a policy the use *can* schedule the update for it. 
Also this requires an admin page for policy in UI. I think it's worth expanding/outlining better the dependencies on UI at the end.

Reply:
This can be handled at UI/CLI level. For UX, yes, the behaviour should be the one you mentioned. I'll add a comment about it.